File: Updog.Application/Comment/Common/CommentInfo.cs
namespace Updog.Application {
    /// <summary>
    /// Information about a comment.
    /// </summary>
    public sealed class CommentInfo {
        #region Properties
        /// <summary>
        /// It's unique identifier.
        /// </summary>
        public int Id { get; }

        /// <summary>
        /// The username of who created the comment.
        /// </summary>
        public string Author { get; }

        /// <summary>
        /// The text of the comment.
        /// </summary>
        public string Body { get; }
        #endregion

        #region Constructor(s)
        /// <summary>
        /// Create a new comment info.
        /// </summary>
        /// <param name="id">The ID of the comment.</param>
        /// <param name="author">The name of the author.</param>
        /// <param name="body">The body of the comment.</param>
        public CommentInfo(int id, string author, string body) {
            Id = id;
            Author = author;
            Body = body;
        }
        #endregion
    }
}

File: Updog.Client/src/comment/common/comment-info.ts
import { DateUtils } from '@/core/utils/date-utils';

/**
 * Information about a comment.
 */
export class CommentInfo {
    /**
     * Create a new post info.
     * @param id The ID of the post.
     * @param author The author's name.
     * @param body The body (URL or text) of the post.
     * @param date The date of commenting.
     */
    public constructor(public id: number, public author: string, public body: string, public date: Date) {}

    /**
     * Get the time since it was commented in a user friendly
     * readable string.
     */
    public getDifferenceDate() {
        return DateUtils.getDifferenceFromToday(this.date);
    }
}
